source/Ship.cpp

```diff
@@ -735,8 +736,13 @@ string Ship::FlightCheck() const
 		return "limited turn?";
 	if(energy - .8 * solar < .2 * (turnEnergy + thrustEnergy))
 		return "solar power?";
-	if(!hyperDrive && !jumpDrive && !canBeCarried)
-		return "no hyperdrive?";
+	if(!canBeCarried)
+	{
+		if(!hyperDrive && !jumpDrive)
+			return "no hyperdrive?";
+		if(fuel < JumpFuel(hyperspaceSystem))
```

Comment:
`JumpFuel()`

Per my earlier comment:
> You also should not supply a destination system. Note that `hyperspaceSystem` is only populated while accelerating into a hyperspace jump, so in practice this will be like supplying no destination, but it certainly appears otherwise.